File: tests/python/cases/case_080_route_ledger.py
"""
Task case 080: route ledger payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_080_route_ledger(input_data):
    # Task: route ledger records and prepare transport-ready payload.
    task_name = 'route_ledger'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ledger',
        'stage': 'route',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']

File: tests/nodejs/cases/case_080_route_ledger.js
/**
 * Task case 080: route_ledger
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case080RouteLedger(input) {
  // Task: route ledger records and prepare transport-ready payload.
  const taskName = 'route_ledger';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ledger',
    stage: 'route',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case080RouteLedger,
};
